test.c

```diff
@@ -1,12 +1,107 @@
-
 #include "common.h"
+#include "assert.h"
+
+struct Tuple* createExampleTable(int count);
+bool destroyExampleTable(struct Tuple*, int count);
+void testStore(void);
+void testRetrieve(void);
+void testParseGrammer(void);
+
+struct ParseTree* parseGrammer(char* sql) {
+  return NULL;
+}
+
+void store(struct Tuple* tuples, int count) {
+  return;
+}
+
+struct Tuple* retrieve(struct ParseTree tree) {
+  return NULL;
+}
+
+void testParseGrammer(void) {
+  struct ParseTree* parseTree = parseGrammer("SELECT id from table1");
+
+  assert(parseTree != 0, "Should not contain a null pointer!");
+  assert(parseTree->commandType == SELECT, "Command type should equal select!");
+  assert(strcmp("table1", parseTree->table) == 0, "The table was not set propertly!");
+  assert(parseTree->whereConstraints != NULL, "WhereConstraints should have been specified!");
+  assert(parseTree->updateFields == NULL, "UPDATE fields were not specified!");
+  assert(parseTree->insertFields == NULL, "INSERT fields were not specified!");
+  assert(parseTree->fields != NULL, "Field should not be null!");
+}
+
+void testStore(void) {
+}
 
+void testRetrieve(void) {
+}
+
+/*
+ * No comment! <-- This is the most ironic thing ever!
+ * Seriously though, this is just the main method!
+ */
 int main(void) {
-  struct ParseTree* p = (struct ParseTree*) malloc(PARSETREE_SIZE);
-  p->commandType = SELECT;
-  p->table = "table1";
-  p->fields = malloc(FIELD_SIZE * 10);
+  printf("===============Example Table:\n");
 
-  printf("%s\n", p->table);
+  // create a table with 10 tuples.
+  struct Tuple* table = createExampleTable(10);
+  destroyExampleTable(table, 10);
```

Comment:
So passing around this count variable everywhere is kinda super annoying. We could encapsulate this stuff into a table structure.